### PR TITLE
Replaced existing DID resolution library with `@web5/dids`

### PR DESCRIPTION
**Pull request by thehenrytsai**
- Replaced existing DID resolution library with `@web5/dids`
- Added `did:dht` support

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,7 +91,8 @@ DWN SDK includes a polyfilled distribution that can imported in a `module` scrip
     const dwn = await Dwn.create({ messageStore, dataStore, eventLog });
 
     // Generate a did:key DID (Decentralized Identifier).
-    const didKey = await DidKeyResolver.generate();
+    // TODO:
```
Comment (LiranCohen) [inline README.md:94]: missing TODO?

File: src/handlers/events-query.ts [hunk 2]
```diff
@@ -32,8 +32,8 @@ export class EventsQueryHandler implements MethodHandler {
       return messageReplyFromError(e, 401);
     }
 
-    const logFilters = Events.convertFilters(message.descriptor.filters);
-    const { events, cursor } = await this.eventLog.queryEvents(tenant, logFilters, message.descriptor.cursor);
+    const eventFilters = Events.convertFilters(message.descriptor.filters);
```
Comment (LiranCohen) [inline src/handlers/events-query.ts:35]: 🙏

File: tests/handlers/permissions-request.spec.ts [hunk 3]
```diff
@@ -79,8 +78,8 @@ export function testPermissionsRequestHandler(): void {
       });
 
       it('should return 401 if auth fails', async () => {
-        const alice = await DidKeyResolver.generate();
-        alice.keyId = 'wrongValue'; // to fail authentication
+        const alice = await TestDataGenerator.generateDidKeyPersona();
+        // alice.keyId = 'wrongValue';
```
Comment (LiranCohen) [inline tests/handlers/permissions-request.spec.ts:82]:
```suggestion
```

**Review by LiranCohen — APPROVED** [review]
🐐

File: README.md (hunk 1 above)
Comment (thehenrytsai) [reply README.md:94]: Thanks, forgot to remove the self-reminder after fixing!

File: tests/handlers/permissions-request.spec.ts (hunk 3 above)
Comment (thehenrytsai) [reply tests/handlers/permissions-request.spec.ts:82]: Thanks, forgot to remove after fixing!

**Review by LiranCohen — APPROVED** [review]
🚀

**Pull request merged**